packages/transformers/pug/src/PugTransformer.js

```diff
@@ -0,0 +1,35 @@
+// @flow
+
+import path from 'path';
+import {Transformer} from '@parcel/plugin';
+
+export default new Transformer({
+  async getConfig({asset}) {
+    const config =
+      (await asset.getConfig(['.pugrc', '.pugrc.js', 'pug.config.js'])) || {};
+    return config;
+  },
+
+  async transform({asset, localRequire, config}) {
+    if (!config) {
+      return [asset];
+    }
+
+    const pug = await localRequire('pug', asset.filePath);
+    const html = pug.compileFile(asset.filePath, {
```

Comment:
I am wondering if you can use `compileClientWithDependenciesTracked`? witch return object contains dependencies:
```
{
  'body': 'function (locals) {...}',
  'dependencies': ['filename.pug']
}
```
So you can use this peace of code from `PugAsset` parcel v1:
```
      if (dependencies) {
        for (let item of dependencies) {
          asset.addDependency(item, {
            includedInParent: true
          });
        }
      }
```